lib/Backend/IRBuilder.cpp

```diff
@@ -441,7 +441,7 @@ IRBuilder::Build()
     // Note that for generators, we insert the bailout after the jump table to allow
     // the generator's execution to proceed before bailing out. Otherwise, we would always
     // bail to the beginning of the function in the interpreter, creating an infinite loop.
-    if (m_func->IsJitInDebugMode() && !this->m_func->GetJITFunctionBody()->IsCoroutine())
+    if (m_func->IsJitInDebugMode() && (!this->m_func->GetJITFunctionBody()->IsCoroutine() || this->IsLoopBody()))
```

Comment:
Different debug break point for Generator functions BUT not relevant for LoopBodies.